socs/agents/hwp_supervisor/agent.py

```diff
@@ -1987,6 +2004,13 @@ def disable_driver_board(self, session, params):
         action.sleep_until_complete(session=session)
         return action.success, f"Completed with state: {action.cur_state_info.state}"
 
+    def cancel_shutdown(self, session, params):
+        """cancel_shutdown()
+        **Task** - Cancels shutdown mode
+        """
+        self.shutdown_mode = False
+        return True, "Cancelled shutdown mode"
+
```

Comment:
I take it this is for cancelling a shutdown before the timeout expires. It's a bit unclear in the current implementation. It looks like I could 'cancel' even after a 'stop' is called, though then the driver board will still be disabled.

You might consider adding a separate attribute to track if you've entered the `elif action == 'no_data' and self.shutdown_no_data_timeout >= 0:` check and only allow the `cancel_shutdown` task to be called successfully if we're waiting on that timeout.

Reply:
Ah, and so the user is expected to 'cancel' and then manually re-enable the driver board?